docs/llm/reference.md

```diff
@@ -12,8 +12,27 @@ It is now integrated into OpenVINO Model Server providing efficient way to run g
 
 Check out the [quickstart guide](quickstart.md) for a simple example that shows how to use this feature.
 
+## Servable Types
+
+Starting with 2025.1, we can highlight four servable types. Such distinction is made based on the input type and underlying GenAI pipeline.
+The servable types are:
+- Language Model Continuous Batching,
+- Language Model Stateful,
+- Visual Language Model Continuous Batching,
+- Visual Language Model Stateful.
+
+First part - Language Model / Visual Language Model - determines whether servable accepts only text or both text and images on the input.
+Seconds part - Continuous Batching / Stateful - determines what kind of GenAI pipeline is used as the engine. By default CPU and GPU devices work on Continuous Batching pipelines. NPU device works only on Stateful servable type.
+
+User does not have to explicitly select servable type. It is inferred based on model directory contents and selected target device.
+Model directory contents determine if model can work only with text or visual input as well. As for target device, setting it to `NPU` will always pick Stateful servable, while any other device will result in deploying Continuous Batching servable. 
+
+Stateful servables ignore most of the configuration used by Continuous Batching, but this will be mentioned later. Some servable types have additional limitations mentioned in the limitations section at the end of this document.
+
+Despite all the differences, all servable types share the same LLM calculator which imposes certain flow in every GenAI-based endpoint.
+
 ## LLM Calculator
-As you can see in the quickstart above, big part of the configuration resides in `graph.pbtxt` file. That's because model server text generation servables are deployed as MediaPipe graphs with dedicated LLM calculator that works with latest [OpenVINO GenAI](https://github.com/openvinotoolkit/openvino.genai/tree/master/src/cpp/include/openvino/genai) library. The calculator is designed to run in cycles and return the chunks of responses to the client.
+As you can see in the quickstart, big part of the configuration resides in `graph.pbtxt` file. That's because model server text generation servables are deployed as MediaPipe graphs with dedicated LLM calculator that works with latest [OpenVINO GenAI](https://github.com/openvinotoolkit/openvino.genai/tree/master/src/cpp/include/openvino/genai) library. The calculator is designed to run in cycles and return the chunks of responses to the client.
```

Comment:
dont refer to master, this documentation will stop working when genai change repo structure